share/root.go

```diff
@@ -87,3 +88,43 @@ func MustDataHashFromString(datahash string) DataHash {
 	}
 	return dh
 }
+
+// IsOutsideRange checks if the namespace is outside the min-max range of the given hashes.
+func IsOutsideRange(namespace gosquare.Namespace, leftHash, rightHash []byte) bool {
+	if len(leftHash) < gosquare.NamespaceSize || len(rightHash) < 2*gosquare.NamespaceSize {
```

Comment:
We should add an error to this function for the caller to understand when there's an error bc the caller could interpret this as a success case since 

if !IsOutsideRange is the call, then invalid length hashes could falsely pass that condition.

Reply:
This goes the same for the other functions here.